src/ParallelAlgorithms/DiscontinuousGalerkin/InitializeMortars.hpp

```diff
@@ -0,0 +1,179 @@
+#include "Domain/Tags.hpp"
+#include "Evolution/Conservative/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/FluxCommunicationTypes.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/MortarHelpers.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp"
+#include "ParallelAlgorithms/Initialization/MergeIntoDataBox.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Frame {
+struct Inertial;
+}  // namespace Frame
+namespace Parallel {
+template <typename Metavariables>
+class ConstGlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+namespace dg {
+namespace Actions {
+/*!
+ * \ingroup InitializationGroup
+ * \brief Initialize mortars between elements for exchanging fluxes
+ *
+ *  If the template parameter `AddFluxBoundaryConditionMortars`
+ *  is set to `false` then the mortar data for flux boundary conditions are not
+ *  initialized and other boundary conditions can be applied.
+ *
+ *  Uses:
+ *  - Metavariables:
+ *    - `temporal_id`
+ *    - `local_time_stepping`
+ *  - System:
+ *    - `volume_dim`
+ *  - DataBox:
+ *    - `Tags::Element<Dim>`
+ *    - `Tags::Mesh<Dim>`
+ *    - `temporal_id`
+ *    - `Tags::Interface<Tags::InternalDirections<Dim>, Tags::Mesh<Dim - 1>>`
+ *    - `Tags::Interface<
+ *    Tags::BoundaryDirectionsInterior<Dim>, Tags::Mesh<Dim - 1>>`
+ *
+ *  DataBox changes:
+ *  - Adds:
+ *    - Tags::VariablesBoundaryData
+ *    - Tags::Mortars<Tags::Next<temporal_id_tag>, dim>
+ *    - Tags::Mortars<Tags::Mesh<dim - 1>, dim>
+ *    - Tags::Mortars<Tags::MortarSize<dim - 1>, dim>
+ *  - Removes: nothing
+ *  - Modifies: nothing
+ */
+template <typename Metavariables, bool AddFluxBoundaryConditionMortars = true>
+struct InitializeMortars {
+ private:
+  static constexpr size_t dim = Metavariables::system::volume_dim;
+  using temporal_id_tag = typename Metavariables::temporal_id;
+  using flux_comm_types = dg::FluxCommunicationTypes<Metavariables>;
+  using mortar_data_tag = tmpl::conditional_t<
+      Metavariables::local_time_stepping,
+      typename flux_comm_types::local_time_stepping_mortar_data_tag,
+      typename flux_comm_types::simple_mortar_data_tag>;
+
+ public:
+  using initialization_option_tags = tmpl::list<::Tags::InitialExtents<dim>>;
+
+  template <typename DataBox, typename... InboxTags, typename ArrayIndex,
+            typename ActionList, typename ParallelComponent,
+            Requires<db::tag_is_retrievable_v<::Tags::InitialExtents<dim>,
+                                              DataBox>> = nullptr>
+  static auto apply(DataBox& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    using simple_tags =
+        db::AddSimpleTags<mortar_data_tag,
```

Comment:
Can this be inlined?  It looks like it is only used once and that use is 60 lines from here, so I think the split hurts readability.

Reply:
I like these lists in initializers because they give an overview of what is being initialized. I have placed them at the beginning of the initializer's `apply` function so far because that's nicely visible for someone reading the code. It basically says "these are the tags that are being initialized here". Would you prefer I move them right before the `merge_into_databox` call where they are used? (keeping in mind that makes them less visible)

Reply:
I would prefer that they were moved down, yes.  (Leaving them not-inlined is OK.)  The problem I have with the current form is that it obscures what the different tags are being initialized with.  In order to understand the `merge_into_databox` call at the bottom of the function I have to compare the arguments with this type list.

You will still have a summary at the top, because it is in the documentation.

Originally many of these had to be factored out because they were needed to calculate the type of the initial DataBox.  Personally, I think they hurt readability and would prefer we got rid of them now that they are no longer needed, but I don't know if that is a general opinion.

Reply:
I've had a few times where having the compute item list factored (I had it on the line right before `merge_into_databox`) and found this useful for reducing the amount of info in the `merge_into_databox` line and making most of that be the simple tags. I've also had the issue of having to go hunting for the `simple_tags` type alias, so I'd prefer both of them be right before the `merge_into_databox` call or to inline the `simple_tags` one. But that's just what I've found personally while working with the initializers.

Reply:
ok I agree that inlining the `simple_tags` improves readability of the `merge_into_databox` call. The overview of what tags are being added should be in the documentation. So let's inline the `simple_tags` and put the `compute_tags` in the line before `merge_into_databox` so that it doesn't obstruct readability.